qemu/tests/pvpanic_event_check.py

```diff
@@ -0,0 +1,57 @@
+import aexpect
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check the event after trigger a crash in VM
+
+    1) boot up guest with pvpanic device
+    2) setup crash_kexec_post_notifiers to 1 in VM and reboot VM
+    3) check kdump server in guest
+    4) trigger a crash in guest
+    5) check the event
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Boot guest with pvpanic device", test.log.info)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    check_kdump_service = params["check_kdump_service"]
+    setup_guest_cmd = params["setup_guest_cmd"]
+    check_kexec_cmd = params["check_kexec_cmd"]
+    expect_event = params["expect_event"]
+    trigger_crash_cmd = params["trigger_crash_cmd"]
+
+    error_context.context("Setup crash_kexec_post_notifiers=1 in guest",
+                          test.log.info)
+    session.cmd(setup_guest_cmd)
+    session = vm.reboot(session)
+    s, o = session.cmd_status_output(check_kexec_cmd)
+    if s or o is None:
+        test.error("Failed to setup crash_kexec_post_notifiers in guest")
+
+    error_context.context("Check kdump server status in guest", test.log.info)
+    o = session.cmd_output(check_kdump_service)
+    if o == "inactive":
+        test.error("Kdump service is not active in guest %s" % o)
+
+    error_context.context("Trigger a crash in guest and check qmp event",
+                          test.log.info)
+    try:
+        session.cmd(trigger_crash_cmd, timeout=5)
+    except aexpect.ShellTimeoutError:
+        pass
+    else:
+        test.fail("Guest should crash.")
+    finally:
+        if vm.monitor.get_event(expect_event) is None:
+            test.fail("Not found expect event: %s" % expect_event)
```

Comment:
Is it available to print the output of QMP if no crashloaded displayed? Eg. PVPANIC_PANICKED .etc.

Reply:
We can check the qmp related logs in qmp***.log file.

Reply:
I see. I think that displaying a live QMP monitor is quite straightforward to debug. But it is also LGTM now. : )

Reply:
Okay,thanks!